---
title: "PCA preparation"
output: html_notebook
---



### Get genotype likelihoos in beagle format
```{bash}
~/angsd/angsd \
-bam all_samples.file.list \
-GL 2 \
-out data \
-nThreads 12 \
-doGlf 2 \
-doMajorMinor 1 \
-doMaf 2 \
-SNP_pval 1e-6 \
-minMapQ 30 \
-minQ 20
```


### Convert beagle format into the covariant matrix for R plotting

```{bash}
python /home/jgoncal1/pcangsd/pcangsd.py 
-beagle raw_all_samples_forPCA.beagle.gz \
-o raw_covariants \
-threads 10

```


##### packages and necessary libraries
```{r}
library(data.table)
library(RcppCNPy)
library("factoextra")
library('ggfortify')
library('cowplot')
library('dplyr')
library(ggpubr)
library('patchwork')
library(wesanderson)
library(ggsci)

nice_layout<-   theme_cowplot()+
    panel_border(color = "grey85", size = 1, linetype = 1,
  remove = FALSE, "black") 

# Stetter Lab colors
#33A02C caudatus
#A6CEE3 cruentus
#FF7F00 hybridus
#FB9A99 hypochondriacus
#CAB2D6 quitensis

minhas_cores<-c(rep('#33A02C', 28), rep('#A6CEE3', 21), rep('#FF7F00', 9),  rep('#FB9A99', 18), rep('#CAB2D6', 12))
```



### Plot PCA

```{r}

### load covaraiant file generated from the beagle file
covaraiants<- fread('/Users/josedias/mount/projects/ag-stetter/jdias/projects/Genome_browser/data/covariantes_data.cov')

### Species groups
species<-c(rep('caudatus', 28), rep('cruentus', 21), rep('hypochondriacus', 18), rep('hybridus_CA', 5), rep('hybridus_SA', 4), rep('quitensis', 12))

#Calulation of principal components
PCS<-prcomp(covaraiants)


## calculate explained variance
mx<-prcomp(covaraiants) 
vars <- apply(mx$x, 2, var)
Variation<-(vars/sum(vars))*100
varaibles<-Variation[1:10]


#### PCA plotting 
pc1_2<- ggplot(PCS, aes(PCS$x[,1],PCS$x[,2]))+
    geom_point(aes(color=Species),  shape = 19, size = 3 , alpha = 0.55) +
  xlab("PC 1 (65.16%)") +
  ylab("PC 2 (28.82%)") +
  nice_layout +
  # theme(legend.title = element_text(size = 15),
  #       legend.text = element_text(size = 15))
  scale_fill_manual(values=c('#33A02C', '#A6CEE3', '#FF7F00', '#FB9A99', '#CAB2D6')) +
  scale_color_manual(values=c('#33A02C', '#A6CEE3', '#FF7F00', '#FB9A99', '#CAB2D6'))


pc2_3<- ggplot(PCS, aes(PCS$x[,1],PCS$x[,3]))+
      geom_point(aes(color=Species), shape = 19, size = 3, , alpha = 0.55)  +
  xlab("PC 1 (65.16%)") +
  ylab("PC 3 (1.62%)") +
  nice_layout +
  scale_fill_manual(values=c('#33A02C', '#A6CEE3', '#FF7F00', '#FB9A99', '#CAB2D6')) +
  scale_color_manual(values=c('#33A02C', '#A6CEE3', '#FF7F00', '#FB9A99', '#CAB2D6'))


# side-by-side arranging
pc_plots<-plot_grid(pc1_2+ theme(legend.position="none"),
          pc1_3 + theme(legend.position="none"),
            labels = c("A", "B"), align="vh",
          axis = "b"
          # rows = 2
          # rel_widths = c(1.01, 1)
) 
          


```

